packages/keystone/databaseAdapters/ReplicaKnexAdapter.js

```diff
@@ -0,0 +1,115 @@
+const { KnexAdapter } = require('@keystonejs/adapter-knex')
+const { knex } = require('knex')
+const { get, omit } = require('lodash')
+
+const { getLogger } = require('@open-condo/keystone/logging')
+
+
+const logger = getLogger('replicaKnexAdapter')
+
+class ReplicaKnexAdapter extends KnexAdapter {
+    constructor (props) {
+        super(omit(props, ['connection']))
+        this.readConnection = get(props, ['connection', 'read'])
+        this.writeConnection = get(props, ['connection', 'write'])
+    }
+
+    async _connect () {
+        this.knex = knex({
+            client: 'postgres',
+            pool: { min: 0, max: 1 },
+            connection: this.writeConnection,
+        })
+
+        this.knexMaster = knex({
+            client: 'postgres',
+            pool: { min: 0, max: 3 },
+            connection: this.writeConnection,
+        })
+
+        this.knexRead = knex({
+            client: 'postgres',
+            pool: { min: 0, max: 3 },
+            connection: this.readConnection,
+        })
+
+        this.knex.context.transaction = (...props) => {
+            return this.knexMaster.context.transaction(...props)
+        }
+
+        const checkUseMasterSingle = (object) => {
+            // if object.method equals "insert", "del" or "update" then use master endpoint
+            if (object.method !== undefined) {
+                return ['insert', 'del', 'update'].includes(object.method)
```

Comment:
what about something like `alter`?

Reply:
Yup, that's possible operation to add. I need to review that configuration